workers/data_refinery_workers/downloaders/geo.py

```diff
@@ -146,187 +147,144 @@ def _download_file_aspera(download_url: str,
-    """Returns the actual file from the archive if it should be queued.
-
-    If the file has been processed or has an unstarted DownloaderJob,
-    None will be returned.
-
-    `extracted_subfile` should be a Dict containing metadata about the
-    file that was extracted from an archive.
-
-    `original_file` should be the file associated with the CURRENT
-    DownloaderJob.
-
-    `samples` are the samples that the actual file should be associated
-    with if it has to be created.
-    """
-    # Check to see if we've made this original file before:
-    potential_existing_files = OriginalFile.objects.filter(
-        source_filename=original_file.source_filename,
-        filename=extracted_subfile['filename'],
-        is_archive=False
-    )
-    if potential_existing_files.count() > 0:
-        # We've already created this record, let's see if we actually
-        # needed to download it or if we just got it because we needed
-        # a file in the same archive.
-        actual_file = potential_existing_files[0]
-
-        if actual_file.needs_processing():
-            if not actual_file.is_downloaded:
-                actual_file.is_downloaded = True
-                actual_file.save()
-            return actual_file
-        else:
-            return None
-
-    else:
-        actual_file = OriginalFile()
-        actual_file.is_downloaded = True
-        actual_file.is_archive = False
-        actual_file.absolute_file_path = extracted_subfile['absolute_path']
-        actual_file.filename = extracted_subfile['filename']
-        actual_file.calculate_size()
-        actual_file.calculate_sha1()
-        actual_file.has_raw = True
-        actual_file.source_url = original_file.source_url
-        actual_file.source_filename = original_file.source_filename
-        actual_file.save()
-
-        for sample in samples:
-            original_file_sample_association = OriginalFileSampleAssociation()
-            original_file_sample_association.sample = sample
-            original_file_sample_association.original_file = actual_file
-            original_file_sample_association.save()
-
-        return actual_file
 
 
 def download_geo(job_id: int) -> None:
     """The main function for the GEO Downloader.
 
     Downloads a single tar file containing the files representing
-    samples relating to a single experiement stored in
+    samples relating to a single experiment stored in
     GEO.
     """
     job = utils.start_job(job_id)
+    accession_code = job.accession_code
+    original_file = job.original_files.first()
 
-    file_assocs = DownloaderJobOriginalFileAssociation.objects.filter(downloader_job=job)
-
-    if file_assocs.count() == 0:
+    if not original_file:
         job.failure_reason = "No files associated with the job."
-        logger.error(
-            "Error occured while extracting tar file.", downloader_job=job_id)
+        logger.error("Error occurred while extracting tar file.", downloader_job=job_id)
```

Comment:
This isn't really true, it looks like we haven't even downloaded the file like alone tried to extract it.